tests/python_package_test/test_sklearn.py

```diff
@@ -1561,6 +1585,52 @@ def test_validate_features(task):
     model.predict(df2, validate_features=False)
 
 
+# LightGBM's 'predict_disable_shape_check' mechanism is intentionally not respected by
+# its scikit-learn estimators, for consistency with scikit-learn's own behavior.
+@pytest.mark.parametrize("task", all_tasks)
+@pytest.mark.parametrize("predict_disable_shape_check", [True, False])
+def test_predict_rejects_inputs_with_incorrect_number_of_features(predict_disable_shape_check, task):
+    X, y, g = _create_data(task, n_features=4)
+    model_factory = task_to_model_factory[task]
+    fit_kwargs = {"X": X[:, :-1], "y": y}
+    if task == "ranking":
+        estimator_name = "LGBMRanker"
+        fit_kwargs.update({"group": g})
+    elif task == "regression":
+        estimator_name = "LGBMRegressor"
+    else:
+        estimator_name = "LGBMClassifier"
```

Comment:
```suggestion
    estimator_name = model_factory.__class__.__name__
    if task == "ranking":
        fit_kwargs.update({"group": g})
```

Reply:
Using string literals here was an intentional choice. I know it makes this a bit more verbose, but I thought that was worth it in exchange for a stricter test.

The code I'm introducing in the library also uses `_estimator.__class__.__name__`. I was worried that if that access pattern stops working in the future (like if `__name__` returns `None`), then this test wouldn't catch the change in the error message if it also relied on `__name__`.

Given that, do you still think we should consolidate to using `__name__`, to make the test easier to read? I'll defer to your opinion, I don't feel strongly.

Reply:
Makes sense!